proto/regen/ecocredit/v1beta1/state.proto

```diff
@@ -0,0 +1,199 @@
+    string project_id = 1;
+
+    // class_id is the unique ID of credit class for this project.
+    string class_id = 2;
+
+    // issuer is the issuer of the credit batches for this project.
+    string issuer = 3;
+
+    // project_location is the location of the project.
+    // Full documentation can be found in MsgCreateProject.project_location.
+    string project_location = 4;
+
+    // metadata is any arbitrary metadata attached to the project.
+    bytes metadata = 5;
+}
+
+// BatchInfo represents the high-level on-chain information for a credit batch.
+message BatchInfo {
+
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 4,
+        primary_key: {fields: "id", auto_increment: true}
+        index: {fields: "batch_denom", unique: true}
+        index: {id: 1 fields: "project_id"}
+        index: {id: 2 fields: "start_date"}
+    };
+
+    // id is a an incrementing integer to identify the batch
+    uint64 id = 1;
+
+    // project_id is the unique ID of the project this batch belongs to.
+    string project_id = 2;
+
+    // batch_denom is the unique ID of credit batch.
+    string batch_denom = 3;
+
+    // metadata is any arbitrary metadata attached to the credit batch.
+    bytes metadata = 5;
+
+    // start_date is the beginning of the period during which this credit batch
+    // was quantified and verified.
+    google.protobuf.Timestamp start_date = 6 [ (gogoproto.stdtime) = true ];
+
+    // end_date is the end of the period during which this credit batch was
+    // quantified and verified.
+    google.protobuf.Timestamp end_date = 7 [ (gogoproto.stdtime) = true ];
+}
+
+// CreditType defines the measurement unit/precision of a certain credit type
+// (e.g. carbon, biodiversity...)
+message CreditType {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 5,
+        primary_key: {fields: "abbreviation"}
+    };
+
+    // the type of credit (e.g. carbon, biodiversity, etc)
+    string name = 1;
+
+    // abbreviation is a 1-3 character uppercase abbreviation of the CreditType
+    // name, used in batch denominations within the CreditType. It must be unique.
+    string abbreviation = 2;
+
+    // the measurement unit (e.g. kg, ton, etc)
+    string unit = 3;
+
+    // the decimal precision
+    uint32 precision = 4;
+}
+
+// ProjectSequence stores and increments the sequence number for projects
+message ProjectSequence {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 7,
+        primary_key: {fields: "project_id"}
+    };
+
+    // project_id is the id of the project
+    string project_id = 1;
+
+    // next_id is the sequence number for the project
+    uint64 next_id = 2;
+}
+
+// EcocreditBalance stores each users credit balance
+message EcocreditBalance {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 8,
+        primary_key: {fields: "address,batch_id"}
+    };
+
+    // address is the address of the credit holder
+    string address = 1;
```

Comment:
let's use `bytes`

Reply:
should we be using cosmos scalars? or just change to bytes for now